benchmarks/benchmarks/solarposition.py

```diff
@@ -0,0 +1,43 @@
+"""
+ASV benchmarks for solarposition.py
+"""
+
+import pandas as pd
+import pvlib
+from pvlib import solarposition
+
+from pkg_resources import parse_version
+
+
+if parse_version(pvlib.__version__) >= parse_version('0.6.1'):
+    sun_rise_set_transit_spa = solarposition.sun_rise_set_transit_spa
+else:
+    sun_rise_set_transit_spa = solarposition.get_sun_rise_set_transit
+
+
+class SolarPosition:
+
+    def setup(self):
+        self.times = pd.date_range(start='20180601', freq='1min',
+                                   periods=14400)
+        self.times_localized = self.times.tz_localize('Etc/GMT+7')
+        self.lat = 35.1
+        self.lon = -106.6
+
+    # GH 512
+    def time_ephemeris(self):
+        solarposition.ephemeris(self.times, self.lat, self.lon)
+
+    # GH 512
+    def time_ephemeris_localized(self):
+        solarposition.ephemeris(self.times_localized, self.lat, self.lon)
+
+    def time_spa_python(self):
+        solarposition.spa_python(self.times_localized[::5], self.lat, self.lon)
```

Comment:
@kanderso-nrel any objection to removing the slice? Will it cause a blip in the asv results or will it the test be rerun over the whole commit history? I ask because it makes the tests a little harder to reason about and it seems like the run time would still be reasonable. I also wonder about the slice performance on a pandas index. 

It looks like the `[::30]` below is needed to keep a reasonable run time in that function. I still wonder if we'd be better off with a new index e.g. `self.days = pd.date_range(start='20180101', end='20190101', freq='24h')`

xref https://github.com/SolarArbiter/solarforecastarbiter-core/issues/567#issuecomment-692253308

Reply:
Yep, those slices were trying to keep runtime to something more than milliseconds and less than tens of seconds.  You raise a good point about the performance of the slice itself -- it was careless to include that in the timing function.  Do you think it is valuable to use the same index for all the functions (meaning their timings can be compared apples-to-apples), or is it better to have separate indexes with various lengths defined in `setup` so that each function runs about the same amount of time?

> Will it cause a blip in the asv results or will it the test be rerun over the whole commit history? 

I *think* asv will automatically invalidate the old results when the function changes (see 'version' [here](https://asv.readthedocs.io/en/stable/benchmarks.html#benchmark-attributes)), but I don't think it would run the new function over the commit history unless you told it to.  In this case I would probably delete the old results and regenerate them with the new function.  It is easy enough to run timings for a single benchmark, it's like pytest where you can specify which functions you want to run. 

Just checking, is there anything special about the index you suggested other than the number of elements?  For example is there a difference between 365 points across a year vs 365 across a single day.

Reply:
Thanks for the info. I can make a PR for this.

> something more than milliseconds

I skimmed the documentation for guidance for a minimum test time and didn't find anything. Milliseconds seems reasonable to me but I'm wondering if you have a reference.

>  Do you think it is valuable to use the same index for all the functions (meaning their timings can be compared apples-to-apples), or is it better to have separate indexes with various lengths defined in setup so that each function runs about the same amount of time?

I vote for apples to apples so long as the runtime constraints are satisfied.

> Just checking, is there anything special about the index you suggested other than the number of elements? 

Nothing special.

Reply:
> Milliseconds seems reasonable to me but I'm wondering if you have a reference

I do not, it just seemed like a good idea at the time.  I see that the [aoi function](https://pvlib-benchmarker.github.io/pvlib-benchmarks/#irradiance.Irradiance.time_aoi) is ~2.5 ms and looks fine, so no pushback from me on having some fast benchmarks 👍 

> I can make a PR for this.

If you want, you could also fix a typo in the [fuentes benchmark](https://github.com/pvlib/pvlib-python/blob/master/benchmarks/benchmarks/temperature.py#L50) where I was inconsistent with `windspeed` vs `wind_speed`.  Didn't show up before the v0.8 release because that benchmark only runs for >=v0.8.0.

Reply:
https://github.com/pvlib/pvlib-python/pull/1059 removes the slices